document.go

```diff
@@ -229,63 +165,39 @@ func (d Document) MarshalBinary() ([]byte, error) {
 	return buf.Bytes(), err
 }
 
-// ETag is type for storing the calculated etag/checksum of Document
-type ETag uint64
-
-func (e ETag) String() string {
-	return fmt.Sprintf("%x", uint64(e))
-}
-
-// NewETag returns new ETag initialized with provided hexadecimal string
-func NewETag(s string) (ETag, error) {
-	var e uint64
-	if _, err := fmt.Sscanf(s, "%x", &e); err != nil {
-		return ETag(e), err
-	}
-	return ETag(e), nil
-}
-
-// ETag returns the checksum of the document
-func (d Document) ETag() (ETag, error) {
-	h := fnv.New64a()
-	err := encodeDocument(h, d, true)
-	return ETag(h.Sum64()), err
-}
-
-// ErrGAPI type represents error message as returned by GAPI
+// ErrGAPI type represents error response returned by GAPI
+//
+// See: https://developers.gadventures.com/docs/rest.html#errors
 type ErrGAPI struct {
-	URI            string
 	HTTPStatusCode int
-	Msg            string
+	Message        string
+	ErrorID        string
+	URI            string
 }
 
+// Error satisfies the error interface for the ErrGAPI type
 func (r *ErrGAPI) Error() string {
-	return fmt.Sprintf("GAPI error for %s HTTP Status %d %s", r.URI, r.HTTPStatusCode, r.Msg)
+	return fmt.Sprintf("GAPI error for %s HTTP Status %d %s %s", r.URI, r.HTTPStatusCode, r.ErrorID, r.Message)
 }
 
-//GAPIError returns ErrGAPI if the document is an GAPI error
-///nil otherwise
+// GAPIError returns *ErrGAPI if the document is a GAPI error, nil otherwise
 func (d Document) GAPIError(uri string) *ErrGAPI {
-	statusCode, hasCode := d["http_status_code"]
-	status, hasStatus := d["status"]
-	message, hasMsg := d["message"]
-	if hasCode && hasMsg {
-		statusCode, hasCode := statusCode.(float64)
-		message, hasMsg := message.(string)
-		if hasCode && hasMsg {
-			return &ErrGAPI{uri, int(statusCode), message}
-		}
-	} else if hasStatus && hasMsg {
-		statusCode, hasCode := status.(float64)
-		message, hasMsg := message.(string)
-		if hasCode && hasMsg {
-			return &ErrGAPI{uri, int(statusCode), message}
+	if _, hasErr := d["error_id"]; hasErr {
+		errID, _ := d["error_id"].(string)
+		msg, _ := d["message"].(string)
+		status, _ := d["http_status_code"].(int)
+		return &ErrGAPI{
+			ErrorID:        errID,
+			HTTPStatusCode: status,
+			Message:        msg,
+			URI:            uri,
```

Comment:
```suggestion
	_, hasErr := d["error_id"]
	if !hasErr {
		return nil
	}
	errID, _ := d["error_id"].(string)
	msg, _ := d["message"].(string)
	status, _ := d["http_status_code"].(int)
	return &ErrGAPI{
		ErrorID:        errID,
		HTTPStatusCode: status,
		Message:        msg,
		URI:            uri,
	}
```

I believe returning early if it's not the "happy path" is a bit more readable.